pkg/logger/ecs_error_log.go

```diff
@@ -0,0 +1,175 @@
+package logger
+
+import (
+	"fmt"
+	"go.uber.org/zap/zapcore"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+)
+
+var ErrorFilePath = os.Getenv("STATUS_MESSAGE_FILE_PATH")
+
+var errorLogMaxAge = os.Getenv("STATUS_MESSAGE_TTL")
+var errorLogMaxAgeInHours, _ = strconv.Atoi(errorLogMaxAge)
+
+var size = os.Getenv("STATUS_MESSAGE_MAX_BYTE_LENGTH")
+var errorFileSize, _ = strconv.Atoi(size)
+
+var queue []zapcore.Entry
+var errorLogChannel = make(chan zapcore.Entry, 10)
+var ticker = time.NewTicker(10 * time.Minute)
+var errorFile *os.File
+
+// ECSErrorReporter func help select the different case between either write error log into file or
+// rotate and delete old logs in file based on TTL
+func ECSErrorReporter() {
+	defer ticker.Stop()
+	for {
+		select {
+		// when new error add in the channel, this case will write it into file
+		case e := <-errorLogChannel:
+			_, _ = WriteError(e, ErrorFilePath)
```

Comment:
I don't love that we are ignoring errors from `WriteError`. I know  may be we can't write an error with the logger since this already processes logging errors and that would lead to recursive calls...

But would it be safe to print the error to stdout? That way it isn't completely ignored. If we ever have any issues with the error logger it would be impossible to debug...

Reply:
Maybe we can write something like log.Printf

Reply:
yeah, will use `log.Printf`

Reply:
I am not sure if you can use `log.Printf`,  because we do set the output for `log`. Though if it is running in container we should be using the default and writes to stderr.

https://github.com/aws-observability/aws-otel-collector/blob/87ebdac1fb521c2d1033538b8161edf935fd9e34/pkg/logger/logger.go#L70-L85

Reply:
Thanks for the callout. I found the collector checked if it is running in container: https://github.com/aws-observability/aws-otel-collector/blob/87ebdac1fb521c2d1033538b8161edf935fd9e34/pkg/logger/logger.go#L41
In this way, I think we are okay to use `log.Printf` if we are running in the container. By the way, I am not sure if running in container is our only use case here.

Reply:
since this ECS error reporter will only run in ECS, i think it will always run in container